spanner/api/Spanner.Samples.Tests/SpannerFixture.cs

```diff
@@ -0,0 +1,223 @@
+            }
+            catch (Exception) { }
+        }
+    }
+
+    private async Task InitializeDatabaseAsync()
+    {
+        // If the database has not been initialized, retry.
+        CreateDatabaseAsyncSample createDatabaseAsyncSample = new CreateDatabaseAsyncSample();
+        InsertDataAsyncSample insertDataAsyncSample = new InsertDataAsyncSample();
+        AddColumnAsyncSample addColumnAsyncSample = new AddColumnAsyncSample();
+        AddIndexAsyncSample addIndexAsyncSample = new AddIndexAsyncSample();
+        AddStoringIndexAsyncSample addStoringIndexAsyncSample = new AddStoringIndexAsyncSample();
+        await createDatabaseAsyncSample.CreateDatabaseAsync(ProjectId, InstanceId, DatabaseId);
+        await insertDataAsyncSample.InsertDataAsync(ProjectId, InstanceId, DatabaseId);
+        await addColumnAsyncSample.AddColumnAsync(ProjectId, InstanceId, DatabaseId);
+        await addIndexAsyncSample.AddIndexAsync(ProjectId, InstanceId, DatabaseId);
+        // Add storing Index on table.
+        await addStoringIndexAsyncSample.AddStoringIndexAsync(ProjectId, InstanceId, DatabaseId);
+        // Update the value of MarketingBudgets.
+        await RefillMarketingBudgetsAsync(300000, 300000);
+    }
+
+    private async Task InitializeBackupAsync()
+    {
+        // Sample database for backup and restore tests.
+        try
+        {
+            CreateDatabaseAsyncSample createDatabaseAsyncSample = new CreateDatabaseAsyncSample();
+            InsertDataAsyncSample insertDataAsyncSample = new InsertDataAsyncSample();
+            await createDatabaseAsyncSample.CreateDatabaseAsync(ProjectId, InstanceId, BackupDatabaseId);
+            await insertDataAsyncSample.InsertDataAsync(ProjectId, InstanceId, BackupDatabaseId);
+        }
+        catch (Exception e) when (e.ToString().Contains("Database already exists"))
+        {
+            // We intentionally keep an existing database around to reduce
+            // the likelihood of test timeouts when creating a backup so
+            // it's ok to get an AlreadyExists error.
+            Console.WriteLine($"Database {BackupDatabaseId} already exists.");
+        }
+
+        try
+        {
+            CreateBackupSample createBackupSample = new CreateBackupSample();
+            createBackupSample.CreateBackup(ProjectId, InstanceId, BackupDatabaseId, BackupId);
+        }
+        catch (RpcException e) when (e.StatusCode == StatusCode.AlreadyExists)
+        {
+            // We intentionally keep an existing backup around to reduce
+            // the likelihood of test timeouts when creating a backup so
+            // it's ok to get an AlreadyExists error.
+            Console.WriteLine($"Backup {BackupId} already exists.");
+        }
+        catch (RpcException e) when (e.StatusCode == StatusCode.FailedPrecondition
+        && e.Message.Contains("maximum number of pending backups (1) for the database has been reached"))
```

Comment:
Does this need to be indented?

Reply:
Yes, As we are not deleting the backup that has been created in test setup because of timeout issue.
When setup runs first time or backup has been expire, it checks if backup has been already in progress on other environment or setup.